app/src/test/java/com/eis/geoCalendar/database/RoomEventDatabaseTest.java

```diff
@@ -0,0 +1,185 @@
+            //Getting the field named "activeInstances" (the map with all active databases)
+            Field instance = AbstractEventDatabase.class.getDeclaredField("activeInstances");
+            //Making it accessible
+            instance.setAccessible(true);
+            //Casting it to a Map
+            Map<?, ?> privateField = (Map<?, ?>) instance.get(null);
+            //Making it empty
+            if (privateField != null)
+                privateField.clear();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Testing whether two instances with the same name are actually the same instance.
+     */
+    @Test
+    public void poolSameName() {
+        AbstractEventDatabase anotherDatabase =
+                AbstractEventDatabase.getInMemoryInstance(RuntimeEnvironment.systemContext,
+                        DEFAULT_DB_NAME);
+        assertEquals(database, anotherDatabase);
+    }
+
+    /**
+     * Testing whether two instances with a different name are different instances.
+     */
+    @Test
+    public void poolDifferentName() {
+        AbstractEventDatabase anotherDatabase =
+                AbstractEventDatabase.getInMemoryInstance(RuntimeEnvironment.systemContext,
+                        "Another name");
+        assertNotEquals(database, anotherDatabase);
+    }
+
+    /**
+     * Testing whether the database actually starts empty.
+     */
+    @Test
+    public void startsEmpty() {
+        final int expectedStartCount = 0;
+        assertEquals(expectedStartCount, database.access().count());
+    }
+
+    /**
+     * Testing single insertion.
+     */
+    @Test
+    public void canAddOne() {
+        final int expectedCount = 1;
+        database.access().insertEntity(new StringEntity(EXAMPLE));
+        assertEquals(expectedCount, database.access().count());
+    }
+
+    /**
+     * Testing the insertion is actually correct, through {@link AbstractStringDao#contains(String)}.
+     */
+    @Test
+    public void canAddOneAndItsCorrect() {
+        database.access().insertEntity(EXAMPLE_ENTITY);
+        assertTrue(database.access().contains(EXAMPLE));
+    }
+
+    /**
+     * Testing multiple insertions.
+     */
+    @Test
+    public void canAddMultiple() {
+        final int expectedCount = 3;
+        database.access().insertEntities(EXAMPLE_COLLECTION);
+        assertEquals(expectedCount, database.access().count());
+    }
+
+    /**
+     * Testing multiple insertions are actually correct, through {@link AbstractStringDao#getAllEntities()}.
+     */
+    @Test
+    public void canAddMultipleAndTheyAreCorrect() {
+        database.access().insertEntities(EXAMPLE_COLLECTION);
+        StringEntity[] allEntities = database.access().getAllEntities();
+        assertEquals(EXAMPLE_COLLECTION.size(), allEntities.length);
+        for (int i = 0; i < EXAMPLE_COLLECTION.size(); i++) {
+            if(!EXAMPLE_COLLECTION.get(i).equals(allEntities[i]))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['if' is not followed by whitespace.](https://app.codacy.com/manual/CremaLuca/geoCalendar/pullRequest?prid=4822079)